internal/action/actions.go

```diff
@@ -160,6 +160,52 @@ func (h *BufPane) Center() bool {
 	return true
 }
 
+// CursorToViewTop moves the cursor to the top of the view,
+// offset by scrollmargin unless at the beginning or end of the file
+func (h *BufPane) CursorToViewTop() bool {
+	v := h.GetView()
+	h.Buf.ClearCursors()
+	scrollmargin := int(h.Buf.Settings["scrollmargin"].(float64))
+	bStart := display.SLoc{0, 0}
+	if v.StartLine == bStart {
+		scrollmargin = 0
+	}
+	h.Cursor.GotoLoc(h.LocFromVLoc(display.VLoc{
+		SLoc:    h.Scroll(v.StartLine, scrollmargin),
+		VisualX: 0,
+	}))
+	return true
+}
+
+// CursorToViewCenter moves the cursor to the center of the view
+func (h *BufPane) CursorToViewCenter() bool {
+	v := h.GetView()
+	h.Buf.ClearCursors()
+	h.Cursor.GotoLoc(h.LocFromVLoc(display.VLoc{
+		SLoc:    h.Scroll(v.StartLine, h.BufView().Height/2),
+		VisualX: 0,
+	}))
+	return true
+}
```

Comment:
Shouldn't we move here from the middle of the screen/view to the first line in the moment we're already at 50% of the first page and move one page up again....vice versa for the move to the bottom?
Just to be consistent with CursorToView{Top|Bottom}?

Reply:
I'm not sure I follow. The idea behind the `CursorTo...` actions is to just move the cursor. I think the action you're looking for already exists, it's called `Center`. It moves the view so that the cursor location becomes the center of the view.